components/Flashcard/Flashcard.js

```diff
@@ -1,19 +1,43 @@
 import styled from "styled-components";
 import { useState } from "react";
 
-export default function Flashcard({ question, answer, collection }) {
+export default function Flashcard({ flashcard, onIsCorrect }) {
   const [showAnswer, setShowAnswer] = useState(false);
+  const [isCorrect, setIsCorrect] = useState(flashcard.isCorrect);
+
+  const { question, answer, collectionTitle: collection, id } = flashcard;
 
   function handleShowAnswer() {
     setShowAnswer(!showAnswer);
   }
 
+  function handleToggleIsCorrect() {
+    setIsCorrect(!isCorrect);
+    onIsCorrect(id);
+  }
+
   return (
     <CardContainer onClick={handleShowAnswer}>
       <StyledFlashcard $showAnswer={showAnswer}>
         <CardFront>
           <CollectionTitle>{collection}</CollectionTitle>
           <Question>{question}</Question>
+          {isCorrect && (
+            <StyledButton
+              $isCorrect={isCorrect}
+              onClick={handleToggleIsCorrect}
```

Comment:
```suggestion
  const { question, answer, collectionTitle: collection, id, isCorrect } = flashcard;

  function handleShowAnswer() {
    setShowAnswer(!showAnswer);
  }

  return (
    <CardContainer onClick={handleShowAnswer}>
      <StyledFlashcard $showAnswer={showAnswer}>
        <CardFront>
          <CollectionTitle>{collection}</CollectionTitle>
          <Question>{question}</Question>
          {isCorrect && (
            <StyledButton
              $isCorrect={isCorrect}
              onClick={() => onIsCorrect(id)}
```

Reply:
see the comment above, you can remove the internal "isCorrect" state and replace it with flashcard.isCorrect.

Reply:
Oh yes, you're absolutely right, we didn't noticed that. 😅  We deleted the unnecessary useState. Thank you.